File: app/src/main/java/com/qhy040404/libraryonetap/data/VolunteerData.java
package com.qhy040404.libraryonetap.data;

import com.google.gson.Gson;

import java.io.Serializable;

public class VolunteerData implements Serializable {
    Gson gson = new Gson();

    private static class Gsondata {
        private int numSameID;
        private int numSameName;
        private double totalDuration;
    }

    public int getSameID(String data) {
        return gson.fromJson(data, Gsondata.class).numSameID;
    }

    public int getSameName(String data) {
        return gson.fromJson(data, Gsondata.class).numSameName;
    }

    public double getTotalHours(String data) {
        return gson.fromJson(data, Gsondata.class).totalDuration;
    }
}

File: app/src/main/java/com/qhy040404/libraryonetap/data/VolunteerData.kt
package com.qhy040404.libraryonetap.data

import com.qhy040404.libraryonetap.utils.extensions.decode

object VolunteerData {
    fun getSameID(data: String) = runCatching {
        data.decode<VolunteerDTO>()?.numSameID!!
    }.getOrDefault(-1)

    fun getSameName(data: String) = runCatching {
        data.decode<VolunteerDTO>()?.numSameName!!
    }.getOrDefault(-1)

    fun getTotalHours(data: String) = runCatching {
        data.decode<VolunteerDTO>()?.totalDuration!!
    }.getOrDefault(-1.0)
}

data class VolunteerDTO(
    val numSameID: Int = 0,
    val numSameName: Int = 0,
    val totalDuration: Double = 0.0,
)
